filebeat/input/filestream/internal/input-logfile/store.go

```diff
@@ -211,6 +211,58 @@ func (s *sourceStore) CleanIf(pred func(v Value) bool) {
 	}
 }
 
+// FixUpIdentifiers copies an existing resource to a new ID and marks the previous one
+// for removal. This method does not check whether the resource belongs to this store
+// beause it is meant to migrate inputs without IDs to inputs with IDs without
+//  duplicating data.
+func (s *sourceStore) FixUpIdentifiers(getNewID func(v Value) (string, interface{})) {
+	s.store.ephemeralStore.mu.Lock()
+	defer s.store.ephemeralStore.mu.Unlock()
+
+	fmt.Println("##################### FixUpIdentifiers prefix", s.identifier.prefix)
+	for key, res := range s.store.ephemeralStore.table {
+		fmt.Println("##################### FixUpIdentifiers key", key)
+		if !s.identifier.MatchesInput(key) {
+			fmt.Println("##################### FixUpIdentifiers, skipping", key)
+			continue
+		}
+
+		if !res.lock.TryLock() {
+			fmt.Println("##################### FixUpIdentifiers cannot lock", key)
```

Comment:
Why would this fail?

Reply:
If something else has already locked the resource `TryLock` returns `false` immediately. When `FixUpdateIdentifiers` is called in `Init`, nothing __should__ be locking the resources in theory. But better to be safe than sorry.

@belimawr I am rather curious why you decided to use `TryLock` instead of `Lock`? If the resource is locked by something else, shouldn't we wait until we get access to fix its ID?

Reply:
Thanks, I think what I really wanted to ask is why we use `TryLock` instead of `Lock`. Why is it OK to fail to obtain the lock and avoid executing the code below?

Reply:
You're totally right @kvch ! I had copied `UpdateIdentifiers` and then did some small modifications to fit my use case without putting much thought on the existing code. I'll fix that and double check the rest of the code.

It's not ok to fail to obtain the lock, we should wait.

Reply:
Fixed on c1492d8ec9